dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,291 @@ private FileFilter getNoBundleXMLFileFilter() {
+
+        APILocator.getRoleAPI().addRoleToUser(roleA,userA);
+        APILocator.getRoleAPI().addRoleToUser(roleB,userB);
+        APILocator.getRoleAPI().addRoleToUser(roleC,userC);
+        APILocator.getRoleAPI().addRoleToUser(roleC,userD);
+
+        final Map<String,Object> filtersMap1 =
+                ImmutableMap.of("dependencies",true,"relationships",true,"excludeClasses","Host,Workflow");
+        final FilterDescriptor filterDescriptor1 =
+                new FilterDescriptor("filterTest1.yml","Filter Test Title 1",filtersMap1,false,roleB.getRoleKey()+','+userC.getUserId());
+
+        publisherAPI.addFilterDescriptor(filterDescriptor1);
+
+        //User A
+        List<FilterDescriptor> filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userA);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User B
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userB);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User C
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userC);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertFalse(filterDescriptors.isEmpty());
+        Assert.assertTrue(filterDescriptors.contains(filterDescriptor1));
+
+        //User D
+        filterDescriptors = publisherAPI.getFiltersDescriptorsByRole(userD);
+        Logger.info(this,filterDescriptors.toString());
+        Assert.assertTrue(filterDescriptors.isEmpty());
+    }
+
+    private void createFilterDescriptor(final String key,final String title,final boolean defaultFilter,Map<String,Object> filtersMap){
+        if(!UtilMethods.isSet(filtersMap)){
+            final List<Object> listExcludeClasses = new ArrayList();
+            listExcludeClasses.add("User");
+            listExcludeClasses.add("OSGI");
+            filtersMap =
+                    ImmutableMap.of("dependencies",true,"relationships",true,"excludeClasses",
+                            listExcludeClasses);
+        }
+        publisherAPI.addFilterDescriptor(new FilterDescriptor(key,title,filtersMap,defaultFilter,"DOTCMS_BACK_END_USER"));
+    }
+
+    /**
+     * This test creates and gets a FilterDescriptor using the key as reference.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_success(){
+        final String key = "TestByKeySuccess.yml";
+        createFilterDescriptor(key,"TestByKeySuccess",true,null);
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey(key);
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(key,filterDescriptor.getKey());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)